src/main/kotlin/com/jaredsburrows/license/LicenseReportTask.kt

```diff
@@ -96,43 +96,24 @@ open class LicenseReportTask : DefaultTask() { // tasks can't be final
 
     // If Android project, add extra configurations
     variantName?.let { variant ->
-      // Add buildType configurations
-      configurations.find { it.name == "compile" }?.let {
-        configurationSet.add(configurations.getByName("${buildType}Compile"))
-      }
-      configurations.find { it.name == "api" }?.let {
-        configurationSet.add(configurations.getByName("${buildType}Api"))
-      }
-      configurations.find { it.name == "implementation" }?.let {
-        configurationSet.add(configurations.getByName("${buildType}Implementation"))
-      }
-
-      // Add productFlavors configurations
-      productFlavors.forEach { flavor ->
-        // Works for productFlavors and productFlavors with dimensions
-        if (variant.capitalize().contains(flavor.name.capitalize())) {
-          configurations.find { it.name == "compile" }?.let {
-            configurationSet.add(configurations.getByName("${flavor.name}Compile"))
-          }
-          configurations.find { it.name == "api" }?.let {
-            configurationSet.add(configurations.getByName("${flavor.name}Api"))
-          }
-          configurations.find { it.name == "implementation" }?.let {
-            configurationSet.add(configurations.getByName("${flavor.name}Implementation"))
-          }
-        }
+      configurations.find { it.name == "${variant}RuntimeClasspath" }?.also {
+        configurationSet.add(it)
       }
     }
 
     // Iterate through all the configurations's dependencies
     configurationSet.forEach { set ->
       if (set.isCanBeResolved) {
-        set.resolvedConfiguration.lenientConfiguration.artifacts.forEach { artifact ->
-          val id = artifact.moduleVersion.id
-          val gav = "${id.group}:${id.name}:${id.version}@pom"
-          configurations.getByName(pomConfiguration).dependencies.add(
-            project.dependencies.add(pomConfiguration, gav)
-          )
+        set.resolvedConfiguration.lenientConfiguration.allModuleDependencies.forEach {
+          if (it.configuration == "runtime") {
```

Comment:
Why only runtime?

Reply:
Again this was based on my observations that some dependencies like 'runtimeElements'  couldn't be resolved due to some ambiguity. But after thinking more about it I'm not convinced to this check. I am not sure if there can be other configurations besides 'runtime' that are fine as well. So I changed the approach based on what google is doing in oss license plugin. Please check the fixup.